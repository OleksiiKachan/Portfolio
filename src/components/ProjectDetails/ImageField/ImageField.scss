@import '../../../assets/style/variables';

.imageField {
  display: flex;
  justify-content: stretch;
  align-items: stretch;

  &__imageSlide {
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    width: 100%;
    height: 100%;
  }

  &__imageShader {
    box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.25);
    width: 100%;
    height: 100%;
    background-color: rgba($color-black, 0.15);
    display: flex;
    align-items: flex-end;
  }
}
