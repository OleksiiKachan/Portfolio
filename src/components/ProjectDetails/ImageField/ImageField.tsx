import React from 'react';
import classNames from 'classnames';
import './ImageField.scss';

/*
    <ImageField
      images=[]
      className=''
    />
*/

type PropsType = {
  images: Array<{ imageUrl: string; imageDescription?: string }>;
  className?: string;
};

export default ({ images, className }: PropsType) => {
  return (
    <div className={classNames('imageField', className)}>
      <div
        style={{ backgroundImage: `url(${images[0].imageUrl})` }}
        className={classNames('imageField__imageSlide')}
      >
        {images[0].imageDescription && (
          <div className={classNames('imageField__imageShader')}>
            <span className={classNames('imageField__imageDescription')}>
              {images[0].imageDescription}
            </span>
          </div>
        )}
      </div>
    </div>
  );
};
