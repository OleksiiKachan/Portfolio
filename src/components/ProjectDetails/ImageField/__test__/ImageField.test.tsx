import React from 'react';
import { shallow } from 'enzyme';
import ImageField from '..';

describe('ImageField', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ImageField
        images={[
          {
            imageUrl: 'imageUrlWithDescription',
            imageDescription: 'description',
          },
          { imageUrl: 'imageUrlWithoutDescription' },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
