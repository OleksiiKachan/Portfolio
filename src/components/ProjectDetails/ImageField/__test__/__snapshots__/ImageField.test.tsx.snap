// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageField Snapshot 1`] = `
<div
  className="imageField"
>
  <img
    className="imageField__imageSlide"
  />
  <div
    className="imageField__imageShader"
  />
</div>
`;
